src/domain/entities/Ref.ts

```diff
@@ -0,0 +1,12 @@
+export type Id = string;
+
+export interface Ref {
+    id: Id;
+}
+
+export interface NamedRef extends Ref {
+    name: string;
+}
+
+export type Code = string;
+export type Identifiable = string;
```

Comment:
An identifiable (id/code/name/shortName, right?) is a DHIS2 implementation detail (I don't think we use that concept in the domain). It then can be moved to the data layer.